### PR TITLE
feat(sourcemaps): Add index size tracking

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR adds metrics that track the size of the url and debug id indexes.

### CODE REVIEW

File: src/sentry/debug_files/artifact_bundle_indexing.py [hunk 1]
```diff
@@ -341,7 +342,20 @@ def to_json(self) -> str:
             "files_by_debug_id": self._files_by_debug_id,
         }
 
-        return json.dumps(json_idx)
+        json_index = json.dumps(json_idx)
+
+        if len(self._files_by_url) == 0:
+            metrics.incr(
+                "artifact_bundle_flat_file_indexing.debug_id_index.size_in_mb",
+                amount=size_in_mb(json_index),
```
Comment (Swatinem) [inline src/sentry/debug_files/artifact_bundle_indexing.py:350]: I believe we have an own metrics type for this histogram-type data. Using a counter for this does not make sense. We would want to know p50, p95, etc…

Comment (iambriccardo) [reply src/sentry/debug_files/artifact_bundle_indexing.py:350]: How do we collect that?

Comment (Swatinem) [reply src/sentry/debug_files/artifact_bundle_indexing.py:350]: `metrics.timing`, I know its a horrible name but it is what it is. Being used for histogram-like data all over the place:

https://github.com/getsentry/sentry/blob/b803cbba8c8de912da92f33d49a3eebd61f359db/src/sentry/models/files/abstractfile.py#L301

Comment (iambriccardo) [reply src/sentry/debug_files/artifact_bundle_indexing.py:350]: Thanks for the heads up, that is a nice cache

File: src/sentry/debug_files/utils.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import sentry_sdk
+
+
+def size_in_mb(string: str) -> int:
```
Comment (Swatinem) [inline src/sentry/debug_files/utils.py:4]: no need to round to MB, just output bytes.

datadog can automatically show you the right unit if you manually edit the metric metadata btw.

**Review by Swatinem — APPROVED** [review]

**Pull request merged**